numerics/fast_fourier_transform_body.hpp

```diff
@@ -0,0 +1,152 @@
+﻿#pragma once
+
+#include "numerics/fast_fourier_transform.hpp"
+
+#include "base/bits.hpp"
+#include "quantities/elementary_functions.hpp"
+#include "quantities/numbers.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_fast_fourier_transform {
+
+using base::BitReversedIncrement;
+using quantities::Angle;
+using quantities::Sin;
+using quantities::si::Radian;
+using quantities::si::Unit;
+namespace si = quantities::si;
+
+// This implementation follows Myrnyy, A Simple and Efficient FFT Implementation
+// in C++, Dr. Dobbs, 2007.
+
+template<int array_size_, int chunk_size_ = array_size_>
+class DanielsonLánczos {
```

Comment:
This class could use a comment (and maybe a citation of *Some improvements in practical Fourier analysis and their application to X-ray scattering from liquids*).

Shouldn’t we have the names of Cooley and Tukey somewhere as well?

Reply:
Added stuff to the bibliography.  Cooley-Tukey (🆒🦃) is for any length (with small prime factors).  Danielson-Lánczos is for powers of 2.